### PR TITLE
153 op support atenrepeatdefault

### PR DESCRIPTION
**Pull request by aarushjain29**

### CODE REVIEW

File: py/torch_migraphx/fx/converters/acc_ops_converters.py [hunk 1]
```diff
@@ -868,6 +868,25 @@ def acc_ops_tile(mgx_module, node, args, kwargs):
...
+def acc_ops_repeat(mgx_module, node, args, kwargs):
+    
+    inp = kwargs["input"]
+    repeats = kwargs["repeats"]
+    
+    bool_output = inp.bool_output
+
+    assert not inp.is_quantized()
+
+    inp_shape = list(inp.shape().lens()) if hasattr(inp.shape(), 'lens') else list(inp.shape())
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/acc_ops_converters.py:881]: did you run into a case where inp was somehow a `torch.Tensor` object? I think `inp_shape = inp.shape().lens()` should be fine here.

File: tests/fx/converters/test_shape_ops_fx.py [hunk 2]
```diff
@@ -254,3 +254,18 @@ def test_bool_shape_ops(op, kwargs):
...
+    ((2, 1, 3), (2, 3, 1)),
+])
+def test_repeat(size, repeat_dims):
+    inp = torch.randn(size)
+    #mod_func = FuncModule(torch.repeat, *repeat_dims)
+    mod_method = MethodModule('repeat', *repeat_dims)
+
+    for mod in [mod_method]:
+        mgx_mod = convert_to_mgx(mod, [inp])
+        verify_outputs(mod, mgx_mod, inp)
```
Comment (shivadbhavsar) [inline tests/fx/converters/test_shape_ops_fx.py:271]:
```suggestion
    mod_method = MethodModule('repeat', *repeat_dims)

    mgx_mod = convert_to_mgx(mod_method, [inp])
    verify_outputs(mod_method, mgx_mod, inp)
```
Seems like there is no functional call, its only implemented as a method

File: py/torch_migraphx/fx/converters/aten_ops_converters.py [hunk 3]
```diff
@@ -97,6 +97,13 @@ def aten_ops_squeeze(mgx_module, node, args, kwargs):
     return acc_ops_converters.acc_ops_squeeze(mgx_module, node, (), acc_kwargs)
 
 
+@migraphx_converter(torch.ops.aten.repeat.default)
+def aten_ops_squeeze(mgx_module, node, args, kwargs):
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/converters/aten_ops_converters.py:101]:
```suggestion
def aten_ops_repeat(mgx_module, node, args, kwargs):
```

**Review by shivadbhavsar — COMMENTED** [review]
Need to add a test case in tests/dynamo/converters/test_shape_ops_dynamo.py

File: tests/fx/converters/test_shape_ops_fx.py (hunk 2 above)
Comment (aarushjain29) [reply tests/fx/converters/test_shape_ops_fx.py]: There is no torch.repeat(...) functional call, so only need to support the "call_method" case.

File: py/torch_migraphx/fx/tracer/acc_tracer/acc_ops.py [hunk 4]
```diff
@@ -352,6 +352,20 @@ def tile(*, input, dims):
...
+@register_acc_op_mapping(
+    op_and_target=("call_method", "repeat"),
+    arg_replacement_tuples=[
+        ("input", "input"),
+        ("*", "repeats"),
+    ],
+)
+
+
+
```
Comment (shivadbhavsar) [inline py/torch_migraphx/fx/tracer/acc_tracer/acc_ops.py:364]:
```suggestion
```
format

File: py/torch_migraphx/fx/converters/acc_ops_converters.py (hunk 1 above)
Comment (shivadbhavsar) [reply py/torch_migraphx/fx/converters/acc_ops_converters.py]: can you respond to this? If you are seeing something here in kwargs["input"] that is not a migraphx insturction, that is concerning

File: tests/fx/converters/test_shape_ops_fx.py [hunk 5]
```diff
@@ -254,3 +254,17 @@ def test_bool_shape_ops(op, kwargs):
 
     mgx_mod = convert_to_mgx(mod, [i1, i2])
     verify_outputs(mod, mgx_mod, [i1, i2])
+
+
+@pytest.mark.parametrize('size, repeat_dims', [
+    ((2, 3), (2, 4)),
+    ((1, 6), (4, 5)),
+    ((2, 1, 3), (2, 3, 1)),
```
Comment (shivadbhavsar) [inline tests/fx/converters/test_shape_ops_fx.py:262]: add a test case here where the repeat_dims are longer than the input size (thats allowed). eg. (2,1,3), (2,1,1,2).

This case should be handled by first unsquezing any missing dimensions in the front of the tensor, and then applying the same logic you already have. Should be a simple enough twek i think. So to be clear the example with `((2,1,3), (2,1,1,2))`, you would first apply an unsqueeze on axis 0. If your repeat dims had 5 values here, then you would have to unsqueze dims 0,1.

File: py/torch_migraphx/fx/converters/acc_ops_converters.py (hunk 1 above)
Comment (shivadbhavsar) [reply py/torch_migraphx/fx/converters/acc_ops_converters.py]: nvm, i see its changed

**Review by shivadbhavsar — CHANGES_REQUESTED** [review]
changing status for clarity

**Review by shivadbhavsar — APPROVED** [review]

**Pull request merged**